Tools/cases_generator/generate_cases.py

```diff
@@ -156,25 +166,17 @@ def write_stack_effect_functions(self) -> None:
         def write_function(
             direction: str, data: list[tuple[AnyInstruction, str]]
         ) -> None:
-            self.out.emit("")
-            self.out.emit("#ifndef NEED_OPCODE_METADATA")
-            self.out.emit(
-                f"extern int _PyOpcode_num_{direction}(int opcode, int oparg, bool jump);"
-            )
-            self.out.emit("#else")
-            self.out.emit("int")
-            self.out.emit(
-                f"_PyOpcode_num_{direction}(int opcode, int oparg, bool jump) {{"
-            )
-            self.out.emit("    switch(opcode) {")
-            for instr, effect in data:
-                self.out.emit(f"        case {instr.name}:")
-                self.out.emit(f"            return {effect};")
-            self.out.emit("        default:")
-            self.out.emit("            return -1;")
-            self.out.emit("    }")
-            self.out.emit("}")
-            self.out.emit("#endif")
+
+            with self.metadata_item(
+                f"int _PyOpcode_num_{direction}(int opcode, int oparg, bool jump)", "", ""
+            ):
+                self.out.emit("    switch(opcode) {")
```

Comment:
You could use ‘with self.out.block()’ and indent() here instead of adding spaces to the statements.